pkg/scaffold/api.go

```diff
@@ -207,18 +213,18 @@ func (api *API) scaffoldV2() error {
 			return fmt.Errorf("error scaffolding kustomization: %v", err)
 		}
 
-		err = crdKustomization.Update()
-		if err != nil {
+		if err = crdKustomization.Update(); err != nil {
 			return fmt.Errorf("error updating kustomization.yaml: %v", err)
 		}
 
 		if !api.resourceExists() {
-			// update scaffolded resource in project file
 			api.project.Resources = append(api.project.Resources,
 				input.Resource{Group: r.Group, Version: r.Version, Kind: r.Kind})
-			err = saveProjectFile("PROJECT", api.project)
-			if err != nil {
-				fmt.Printf("error updating project file with resource information : %v \n", err)
+
+			// If the --force was used to re-crete a resource that was created before then,
+			// the PROJECT file will not be updated.
+			if err = saveProjectFile("PROJECT", api.project); err != nil {
+				return fmt.Errorf("error updating project file with resource information : %v \n", err)
```

Comment:
Why are you changing this behavior? I mean it was printing the error but it was not returning an error so now it will not continue to create the controller and update `main.go` if it is unable to save the PROJECT file while before it was just printing the error but scaffolding the controller.

Reply:
Would make sense continue in this case? 
I think it was just printing the error by mistake. However, your point is valid. 
So, @mengqiy wdyt that would be the correct behaviour in this scenario? Should we return an error when is not possible update the PROJECT file in order to keep the consistency or just print the error in the logs?

Reply:
is the comment above relevant?  My gut says that if we can't save the state of what we're doing to the project file, we should consider that a failure, but I'm not familiar enough with that code.  cc @mengqiy

Reply:
That part is executed after scaffolding the resource related files but before scaffolding the controller related ones, or before updating main.go. Anyway we will end in a incomplete state, so I think that it may be better to let kubebuilder scaffold the controller and update main and log instructions to the user to include the resource into the PROJECT file manually.

Reply:
IHMO the PROJECT file should ensure the consistency of the project because of this I do not think that would make sense continue scaffold if was not possible to add the new resource info into it. However, please @mengqiy and @DirectXMan12 just let me know your final decision to change it if is required for we move forward with this PR.

Reply:
Basically right now we do:

1- Scaffold resource related files
2- Update PROJECT file
3- Scaffold controller related files
4- Update main.go

What about swapping the first two elements?

1- Update PROJECT file
2- Scaffold resource related files
3- Scaffold controller related files
4- Update main.go

This would mean moving all the `if !api.resourceExists() { ... }` block just after checking `if err := api.validateResourceGroup(r); err != nil { return err }`.

Reply:
I don't have a strong opinion about these 2 orders.
It may fail at any step.
@DirectXMan12 @droot thoughts?

Reply:
I like the @Adirio suggestion since if we face some issue to save the PROJECT file than, no scaffold files will be gen.

Reply:
Done as suggested  by @Adirio